src/language-html/preprocess.js

```diff
@@ -176,8 +176,8 @@ function mergeSimpleElementIntoText(ast /*, options */) {
     node.attrs.length === 0 &&
     node.children.length === 1 &&
     node.firstChild.type === "text" &&
-    // \xA0: non-breaking whitespace
-    !/[^\S\xA0]/.test(node.children[0].value) &&
+    // https://infra.spec.whatwg.org/#ascii-whitespace
+    !/[^\t\n\f\r ]/.test(node.children[0].value) &&
```

Comment:
I'm not sure about this part, will find something to test tomorrow, it's late in my location.